test/apps/tangents/app.js

```diff
@@ -0,0 +1,229 @@
+import {AnimationLoop, Model, CylinderGeometry, calculateTangents} from '@luma.gl/engine';
+import {Buffer, clear} from '@luma.gl/webgl';
+import {setParameters} from '@luma.gl/gltools';
+import {Matrix4} from '@math.gl/core';
+import {Vector3, radians} from '@math.gl/core';
+import Controller from './controller';
+
+const vs = `\
+  #define SHADER_NAME simpleVs;
+
+  attribute vec3 positions;
+
+  uniform mat4 u_ModelMatrix;
+  uniform mat4 u_ViewMatrix;
+  uniform mat4 u_ProjectionMatrix;
+
+  void main(void) {
+    gl_Position = u_ProjectionMatrix * u_ViewMatrix * u_ModelMatrix * vec4(positions, 1.0);
+  }
+`;
+
+const fs = `\
+  #define SHADER_NAME simpleFs;
+
+  precision highp float;
+
+  uniform vec4 u_Color;
+
+  void main(void) {
+    gl_FragColor = u_Color;
+  }
+`;
+
+const tangentVs = `\
+  #define SHADER_NAME simpleVs;
+
+  attribute vec3 positions;
+  attribute vec3 colors;
+  attribute vec3 instancePositions;
+  attribute vec3 instanceNormals;
+  attribute vec3 instanceTangents;
+  attribute vec3 instanceBitangents;
+
+  uniform mat4 u_ModelMatrix;
+  uniform mat4 u_ViewMatrix;
+  uniform mat4 u_ProjectionMatrix;
+  
+  varying vec3 vColor;
+
+  void main(void) {
+    vColor = colors;
+    mat3 instanceMatrix = mat3(instanceTangents, instanceBitangents, instanceNormals);
+    
+    vec4 pos = vec4(normalize(instanceMatrix * positions) * 0.2, 1.0);
+    pos.xyz += instancePositions;
+    pos = u_ModelMatrix * pos;
+
+    gl_Position = u_ProjectionMatrix * u_ViewMatrix * pos;
+  }
+`;
+
+const tangentFs = `\
+  precision highp float;
+
+  varying vec3 vColor;
+  void main(void) {
+    gl_FragColor = vec4(vColor, 1.0);
+  }
+`;
+
+const loop = new AnimationLoop({
+  onInitialize({gl}) {
+    this._controller = new Controller(gl.canvas);
+    setParameters(gl, {
+      depthTest: true,
+      depthFunc: gl.LEQUAL,
+      [gl.LINE_WIDTH]: 2
+    });
+
+    const modelMatrix = new Matrix4().rotateX(radians(30)).rotateY(30);
+    const projectionMatrix = new Matrix4();
+
+    const geometry = new CylinderGeometry({
+      radius: 1,
+      nradial: 10,
+      nvertical: 10
+    });
+
+    const {
+      attributes: {NORMAL, TEXCOORD_0, POSITION},
+      indices
+    } = geometry;
+
+    const {tangents, bitangents} = calculateTangents(
```

Comment:
It would be nice if calculateTangents accepted a luma.gl geometry object (looked for attributes and indices)